Consul/Session.cs

```diff
@@ -214,32 +215,29 @@ public Task RenewPeriodic(TimeSpan initialTTL, string id, WriteOptions q, Cancel
                 {
                     throw new ArgumentNullException(nameof(q));
                 }
-                var waitDuration = (int)(initialTTL.TotalMilliseconds / 2);
-                var lastRenewTime = DateTime.Now;
+                var waitDuration = TimeSpan.FromMilliseconds(initialTTL.TotalMilliseconds / 2);
+                var sw = Stopwatch.StartNew();
```

Comment:
Measuring elapsed time with `DateTime` is not reliable, it is better to use `StopWatch`.

Reply:
Agreed